buildconfig/Jenkins/Conda/cppcheck.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash -ex
+
+SCRIPT_DIR=$(dirname "$0")
+# If errors slip through to master this can be used to set a non-zero
+# allowed count while those errors are dealt with. This avoids breaking all
+# builds for all developers
+ALLOWED_ERRORS_COUNT=0
+
+if [[ ${JOB_NAME} == *pull_requests* ]]; then
+    # This relies on the fact pull requests use pull/$PR-NAME
+    # which squashes the branch into a single merge commit
+    cd $WORKSPACE
+
+    if ${SCRIPT_DIR}/check_for_changes cpp; then
+        echo "No C++ files have changed. Skipping check."
+        exit 0
+    fi
+fi
+
+###############################################################################
+# Create the build directory if it doesn't exist
+###############################################################################
+[ -d $WORKSPACE/build ] || mkdir $WORKSPACE/build
+cd $WORKSPACE/build
+
+# remove old results if they exist
+find -name cppcheck.xml -delete
+
+echo $(command -v scl)
+ls /jenkins_workdir/workspace/pull_requests-cppcheck
+ls /usr/bin
+ls /var
+ls /var/jenkins_home
+ls /var/jenkins_home/plugins
+which cmake
+# configure cmake
+if [ $(command -v scl) ]; then
+    CMAKE_EXE=cmake3
+    SCL_ENABLE="scl enable devtoolset-7"
+else
+    CMAKE_EXE=cmake
+    SCL_ENABLE="eval"
+fi
+$SCL_ENABLE "$CMAKE_EXE --version"
+
+if [ "$(command -v ninja)" ]; then
+  CMAKE_GENERATOR="-G Ninja"
+elif [ "$(command -v ninja-build)" ]; then
+  CMAKE_GENERATOR="-G Ninja"
+fi
+$SCL_ENABLE "$CMAKE_EXE ${CMAKE_GENERATOR} -DCMAKE_BUILD_TYPE=Debug -DCPPCHECK_GENERATE_XML=TRUE -DCPPCHECK_NUM_THREADS=$BUILD_THREADS .."
```

Comment:
I was wondering whether it is worth making a new `CMakePreset`, called something like `cppcheck-ci`, so that it's easier to see all of the various configurations rather having to dig through the build scripts?